server/api/personalGoal/__tests__/personalGoal.ability.spec.js

```diff
@@ -0,0 +1,258 @@
+import test from 'ava'
+import request from 'supertest'
+import { server, appReady } from '../../../server'
+import PersonalGoal from '../personalGoal'
+import MemoryMongo from '../../../util/test-memory-mongo'
+
+import Person from '../../person/person'
+import people from '../../person/__tests__/person.fixture'
+import Goal from '../../goal/goal'
+import goals from '../../goal/__tests__/goal.fixture'
+import { PersonalGoalStatus } from '../personalGoal.constants'
+
+import { jwtData, jwtDataDali } from '../../../middleware/session/__tests__/setSession.fixture'
+
+test.before('before connect to database', async (t) => {
+  t.context.memMongo = new MemoryMongo()
+  await t.context.memMongo.start()
+  await appReady
+})
+
+test.after.always(async (t) => {
+  await t.context.memMongo.stop()
+})
+
+test.before('connect and add two personalgoal entries', async (t) => {
+  // connect each personalgoal to an owner and org
+  [t.context.people, t.context.goals] = await Promise.all([
+    Person.create(people),
+    Goal.create(goals)
+  ])
+  // link people to goals
+  const admin = t.context.people[0]
+  const dali = t.context.people[1]
+  const alice = t.context.people[2]
+  const personalGoals = [
+    {
+      person: admin._id,
+      goal: t.context.goals[0]._id,
+      status: PersonalGoalStatus.QUEUED
+    },
+    {
+      person: dali._id,
+      goal: t.context.goals[1]._id,
+      status: PersonalGoalStatus.QUEUED
+    },
+    {
+      person: alice._id,
+      goal: t.context.goals[2]._id,
+      status: PersonalGoalStatus.ACTIVE
+    }
+  ]
+  t.context.personalGoals = await PersonalGoal.create(personalGoals)
+})
+
+// test.after.always(async () => {
+//   await Promise.all([
+//     PersonalGoal.deleteMany(),
+//     Person.deleteMany(),
+//     Goal.deleteMany()
+//   ])
+// })
+
+// ANON
+test('PersonalGoal API - anon - cannot read', async t => {
+  let response = await request(server)
+    .get('/api/personalGoals')
+    .set('Accept', 'application/json')
+  t.is(response.statusCode, 403)
+
+  const id = t.context.personalGoals[1]._id
+  response = await request(server)
+    .get(`/api/personalGoals/${id}`)
+    .set('Accept', 'application/json')
+
+  t.is(response.statusCode, 403)
+})
+
+test('PersonalGoal API - anon - cannot write', async t => {
+  let response = await request(server)
+    .post('/api/personalGoals')
+    .send({
+      person: t.context.people[3]._id,
+      goal: t.context.goals[1]._id
+    })
+
+  t.is(response.statusCode, 403)
+  const id = t.context.personalGoals[1]._id
+  response = await request(server)
+    .get(`/api/personalGoals/${id}`)
+    .send({
+      status: PersonalGoalStatus.ACTIVE
+    })
+
+  t.is(response.statusCode, 403)
+
+  response = await request(server)
+    .get(`/api/personalGoals/${id}`)
+    .send({
+      status: PersonalGoalStatus.ACTIVE
+    })
+
+  t.is(response.statusCode, 403)
+})
+// NORMAL
+test('PersonalGoal API - normal - list self', async t => {
```

Comment:
What does "normal" mean here? Any authenticated user that isn't an admin?

Reply:
yes